qiskit/ignis/experiments/base/experiment.py

```diff
@@ -0,0 +1,180 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Experiment class.
+"""
+
+from typing import Optional
+import uuid
+
+from qiskit import transpile, assemble
+from qiskit.providers import BaseJob
+from qiskit.exceptions import QiskitError
+from qiskit.providers import BaseBackend
+
+from .generator import Generator
+from .analysis import Analysis
+
+
+class Experiment:
+    """Basic experiment class."""
+
+    # pylint: disable=arguments-differ
+    def __init__(self,
+                 generator: Optional[Generator] = None,
+                 analysis: Optional[Analysis] = None,
+                 job: Optional[BaseJob] = None):
+        """Initialize an experiment."""
+
+        self._generator = generator
+        self._analysis = analysis
+        self._job = job
+
+    def run(self, backend: BaseBackend, **kwargs) -> any:
+        """Run an experiment and perform analysis"""
+        self.execute(backend, **kwargs)
+        return self.run_analysis()
+
+    def execute(self, backend: BaseBackend, **kwargs) -> BaseJob:
+        """Execute the experiment on a backend.
+​
+        TODO: Add transpiler options
+
+        Args:
+            backend: backend to run experiment on.
+            kwargs: kwargs for assemble method.
+​
+        Returns:
+            BaseJob: the experiment job.
+        """
+        # Get circuits and metadata
+        exp_id = str(uuid.uuid4())
+        circuits = transpile(self.generator.circuits(),
```

Comment:
Transpiling in this manner prevents the user from controlling the transpilation process, meaning one cannot transpile to a given basis (which affect how simulators handle noise models).

Reply:
In RB, transpiling into basis gates is important in order to calculate the error per gate from the EPC (error per clifford)